### DISCUSSION TITLE
Selector for useMutation result before return

### DISCUSSION THREAD

**Discussion by zirkelc** [TanStack/query#3059] [Ideas] [answered]
I use the `select` option of useQuery to parse JSON strings fields into actual objects before returning them. I think it would be helpful to have the same option available on useMutation. What do you think?

Is there any other workaround to transform the mutation result before returning it?

**Comment by TkDodo** [comment, marked as answer]
at the moment, you can just `await` the result in the `mutateFn`, and then transform it before you return it?

```
useMutation(async () => {
  const { data } = await axios.post(...)
  return transformData(data)
})
```

**Reply by zirkelc** [reply]
is that also possible in the `onSuccess` handler? The promisable return type `Promise<unknown> | void` indicates this.

**Reply by TkDodo** [reply]
You can return a promise there because it will be awaited, but it doesn’t change the type of `data`

**Reply by zirkelc** [reply]
Okay, thank your for the info! Do you guys plan to add `select` for mutations as well?

**Reply by TkDodo** [reply]
my currently vague plan is to bring mutations on-par with queries :)
